### ISSUE TITLE
Using Maui blazor，bind data to MDataTable，throw ”Unable to cast object of type 'Microsoft.AspNetCore.Components.WebView.Services.WebViewJSRuntime' to type 'Microsoft.JSInterop.IJSInProcessRuntime'.“

### ISSUE THREAD

**Issue by dotnet9** [BlazorComponent/MASA.Blazor#533]
**Describe the bug**

我使用Razor类库的方式实现多项目UI共享，供其他项目使用（Blazor Server\Blazor Wasm\Maui Blazor）。其中有使用到MDataTable组件，通过http获取到数据后绑定MDataTable，目前Blazor Server和Blazor Wasm正常运行，只是在Maui blazor中运行时出现异常，异常信息如下：

I use the Razor class library to achieve multi-project UI sharing for other projects (Blazor Server\Blazor Wasm\Maui Blazor). Among them, the MDataTable component is used. After the data is obtained through http api, the MDataTable is bound. At present, Blazor Server and Blazor Wasm are running normally, but an exception occurs when running in Maui blazor. The exception information is as follows:

![maui blazor exception](https://user-images.githubusercontent.com/19390373/176336396-a42c655f-373e-4d5b-8f63-db461ed9972a.png)

```shell
Unable to cast object of type 'Microsoft.AspNetCore.Components.WebView.Services.WebViewJSRuntime' to type 'Microsoft.JSInterop.IJSInProcessRuntime'.
   at BlazorComponent.I18n.CookieStorage.GetCookie(String key)
   at BlazorComponent.I18n.I18n..ctor(CookieStorage cookieStorage, IHttpContextAccessor httpContextAccessor)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass2_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.AspNetCore.Components.ComponentFactory.<>c__DisplayClass7_0.<CreateInitializer>g__Initialize|1(IServiceProvider serviceProvider, IComponent component)
   at Microsoft.AspNetCore.Components.ComponentFactory.PerformPropertyInjection(IServiceProvider serviceProvider, IComponent instance)
   at Microsoft.AspNetCore.Components.ComponentFactory.InstantiateComponent(IServiceProvider serviceProvider, Type componentType)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.InstantiateChildComponentOnFrame(RenderTreeFrame& frame, Int32 parentComponentId)
   at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.InitializeNewComponentFrame(DiffContext& diffContext, Int32 frameIndex)
   at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.InitializeNewSubtree(DiffContext& diffContext, Int32 frameIndex)
   at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.InsertNewFrame(DiffContext& diffContext, Int32 newFrameIndex)
   at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.AppendDiffEntriesForRange(DiffContext& diffContext, Int32 oldStartIndex, Int32 oldEndIndexExcl, Int32 newStartIndex, Int32 newEndIndexExcl)
   at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.ComputeDiff(Renderer renderer, RenderBatchBuilder batchBuilder, Int32 componentId, ArrayRange`1 oldTree, ArrayRange`1 newTree)
   at Microsoft.AspNetCore.Components.Rendering.ComponentState.RenderIntoBatch(RenderBatchBuilder batchBuilder, RenderFragment renderFragment, Exception& renderFragmentException)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.ProcessRenderQueue()
```

服务注册如下：

The service is registered as follows:

```C#
services.AddMasaBlazor();
        services.TryAddScoped<I18n>();
        services.TryAddScoped<CookieStorage>();
        services.AddHttpContextAccessor();
```

界面布局和数据获取部分代码如下：

The interface layout and data acquisition part of the code is as follows:

```html
<MDataTable Headers="_headers"
                    Items="_historyFiles"
                    ItemsPerPage="10"
                    Class="elevation-1"/>

@code{
    ...
    // other code
    ...
    private readonly List<FileInfoDto> _historyFiles = new();
    private async Task QueryFiles(string? subDirectory)
        {
            _historyFiles.Clear();
            var result = await GlobalConfig.BaseApiAddress
                .AppendPathSegments("zh-cn", "file", "list")
                .SetQueryParams(new
                {
                    library = Library,
                    subdirectory = subDirectory
                })
                .GetJsonAsync<ResponseResult<List<FileInfoDto>>>();
            if (result.Data is { Count: > 0 })
            {
                _historyFiles.AddRange(result.Data);
            }
        }
    
    ...
    // other code
    ...
}
```

其他地方调用`QueryFiles`方法，在Maui项目运行时，方法运行完毕抛出上面的异常，大致是这样，您看看这是什么问题，谢谢。

The `QueryFiles` method is called in other places. When the Maui project is running, the above exception is thrown after the method is run. It is roughly like this. You can see what the problem is, thank you.

**Comment by doddgu** [comment]
@capdiem

**Comment by capdiem** [comment]
BlazorComponent/BlazorComponent#162

**Comment by capdiem** [comment]
fixed in 0.5.0.
